pika/adapters/blocking_connection.py

```diff
@@ -700,6 +700,15 @@ def channel(self, channel_number=None):
 
         return channel
 
+    def __enter__(self):
+        # Prepare `with` context
+        return self
+
+    def __exit__(self, *args):
+        # Close connection after `with` context
+        self.close()
+        return self
```

Comment:
BUG: This is the wrong return value for `__exit__`. Returning anything that evaluates to True (like `self` in this case) will suppress the exception. We don't want to suppress the exception. To fix, remove the `return` statement from `__exit__`. See http://effbot.org/zone/python-with-statement.htm